spec/controllers/products_controller_spec.rb

```diff
@@ -0,0 +1,198 @@
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+  before :each do
+    @product = Product.create(name: "starfish", price: 3, merchant_id: 1, description: "A starfish!")
+  end
+
+  let(:create_params) do
+  { merchant_id: 1,
+    product: {
+      name: "Dogfish",
+      price: 2,
+      description: "new description"
+    }
+  }
+  end
+
+  let(:update_params) do
```

Comment:
An alternative to making 3 sets of params here would be to make one valid set and change an attribute in the test to make it different or invalid. You'll follow that pattern when you're using :sparkles:FactoryGirl:sparkles:!